lib/bootstrap/environment.rb

```diff
@@ -64,11 +89,23 @@ def pattern_path(path)
 if $0 == __FILE__
   LogStash::Bundler.setup!({:without => [:build, :development]})
   require ARGV.shift
+
+  if settings = YAML.parse(IO.read("settings.yml"))
+    settings = settings.to_ruby
+    flat_settings_hash = flatten_hash(settings)
+    settings_array = flatten_arguments(flat_settings_hash)
+  else
+    settings_array = []
+  end
+
   # TODO deprecate these arguments in the next major version. use -i only
   if ARGV == ["irb"] || ARGV == ["pry"]
     puts "Warn: option \"#{ARGV.first}\" is deprecated, use \"-i #{ARGV.first}\" or \"--interactive=#{ARGV.first}\" instead"
     LogStash::Runner.run("bin/logstash", ["--interactive", ARGV.first])
   else
-    LogStash::Runner.run("bin/logstash", ARGV)
+    # The Clamp library supports specifying the same argument multiple times
+    # and it keeps the   in an array. So in order for cli args to override
+    # the settings.yml args we can do `settings_array + ARGV`
+    LogStash::Runner.run("bin/logstash", settings_array + ARGV)
```

Comment:
:+1: on giving `settings_array` a better name.
the idea here is to push to the runner both the settings from the yaml file and the settings from the CLI, so I don't see a problem with `+`. Also we can sacrifice performance for readability here since it will only execute once at startup.